javalib/src/main/scala/java/io/DataInputStream.scala

```diff
@@ -20,9 +23,35 @@ class DataInputStream(in: InputStream)
   private final val outBasket = ByteBuffer.wrap(inBasket) // default: BigEndian
 
   private final def rebuffer(n: Int): ByteBuffer = {
-    if (in.read(inBasket, 0, n) < n)
-      throw new java.io.EOFException
-    outBasket.clear() // tricky here: contents preserved, bookkeeping reset.
+    @tailrec
+    def rebufferImpl(n: Int, runningTotal: Int): Int = {
+      in.read(inBasket, 0, n) match {
+        case nRead if (nRead == n) => nRead
```

Comment:
How about simplifying these matches, eg.:
```suggestion
        case `n` => n
```

We can also replace next two with simple `case -1 => ...` and `case 0 => ...`. Looks more readable in my opinion

Reply:
@WojciechMazur  Thank you for the timely & helpful review.  The three changes you suggested do 
improve the code.  I have incorporated all three.

I learned some Scala from your suggestion about using backticks. I never knew that they
could be used in a match to specify an argument reference.  I had always only 
used them to quote 'funny` names.  (The two integer cases were my brain not
being turned on, sorry.)